### PR TITLE
Create law and test: interference minimum condition

### PR DESCRIPTION
**Pull request by Voldoned**

### CODE REVIEW

File: symplyphysics/laws/optics/interference_minimum.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from sympy import (Eq, solve)
+from symplyphysics import (units, Quantity, Symbol, print_expression,
+                           validate_input, validate_output, dimensionless)
+
+# Description
+## If the stroke difference is equal to an odd number of half-waves,
```
Comment (blackyblack) [inline symplyphysics/laws/optics/interference_minimum.py:6]: What's `stroke`? Where can we see it?
What's `stroke difference`? Are they different in shape, size, color or what?

File: symplyphysics/laws/optics/interference_minimum.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## then at this point of the screen there will be a minimum of intensity during interference.
+
+# Law: delta = (2m+1) * lambda / 2
+# Where:
+## delta - optical travel difference for two coherent waves
+## lambda - length of wave
+## m - number of interference minimum
```
Comment (blackyblack) [inline symplyphysics/laws/optics/interference_minimum.py:13]: What's this number?

File: symplyphysics/laws/optics/interference_minimum.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+travel_difference = Symbol("travel_difference", units.length)
+number_minimum = Symbol("number_minimum", dimensionless)
+wave_length = Symbol("wave_length", units.length)
+
+law = Eq(travel_difference, number_minimum * wave_length)
```
Comment (blackyblack) [inline symplyphysics/laws/optics/interference_minimum.py:20]: Does not correspond to description.

File: symplyphysics/laws/optics/interference_minimum.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def print_law():
+    return print_expression(law)
+
+
+@validate_input(wave_length_=wave_length, number_minimum_=number_minimum)
+@validate_output(travel_difference)
+def calculate_travel_difference(wave_length_: Quantity, number_minimum_: Quantity | int) -> Quantity:
```
Comment (blackyblack) [inline symplyphysics/laws/optics/interference_minimum.py:29]:
```suggestion
def calculate_travel_difference(wave_length_: Quantity, number_minimum_: int) -> Quantity:
```

File: symplyphysics/laws/optics/interference_minimum.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    solved = solve(law, wave_length, dict=True)[0][wave_length]
+    result_expr = solved.subs({
+        wave_length: wave_length_,
+        number_minimum: number_minimum_
+    })
+    result = Quantity(result_expr)
+    return result
```
Comment (blackyblack) [inline symplyphysics/laws/optics/interference_minimum.py:36]: Add empty line.

File: test/optics/interference_minimum_test.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    SI,
+    convert_to, prefixes,
+)
+from symplyphysics.laws.optics import interference_minimum as minimum_law
+
+
+@fixture(name="test_args")
+def test_args_fixture():
+    maximum_number = 2
+    wave_length = Quantity(550 * prefixes.nano * units.meters)
```
Comment (blackyblack) [inline test/optics/interference_minimum_test.py:16]: Could you show me, where you've got this example?

File: test/optics/interference_minimum_test.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    Quantity,
+    SI,
+    convert_to, prefixes,
+)
+from symplyphysics.laws.optics import interference_minimum as minimum_law
+
+
+@fixture(name="test_args")
+def test_args_fixture():
+    maximum_number = 2
```
Comment (blackyblack) [inline test/optics/interference_minimum_test.py:15]: It should be minimum number, isn't it?

File: test/optics/interference_minimum_test.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    with raises(errors.UnitsError):
+        minimum_law.calculate_travel_difference(lb, test_args.maximum_number)
+    with raises(TypeError):
+        minimum_law.calculate_travel_difference(100, test_args.maximum_number)
+
+
+def test_bad_number_maximum(test_args):
+    nb = Quantity(1 * units.coulomb)
+    with raises(errors.UnitsError):
+        minimum_law.calculate_travel_difference(test_args.wave_length, nb)
```
Comment (blackyblack) [inline test/optics/interference_minimum_test.py:39]: Also empty line.

File: test/optics/interference_minimum_test.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+def test_bad_wave_length(test_args):
+    lb = Quantity(1 * units.coulomb)
+    with raises(errors.UnitsError):
+        minimum_law.calculate_travel_difference(lb, test_args.maximum_number)
+    with raises(TypeError):
+        minimum_law.calculate_travel_difference(100, test_args.maximum_number)
+
+
+def test_bad_number_maximum(test_args):
```
Comment (blackyblack) [inline test/optics/interference_minimum_test.py:36]: Wrong test case name.

File: symplyphysics/laws/optics/interference_minimum.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+# Description
+## If the travel difference is equal to an odd number of half-waves,
+## then at this point of the screen there will be a minimum of intensity during interference.
+
+# Law: delta = (2m+1) * lambda / 2
+# Where:
+## delta - optical travel difference for two coherent waves
+## lambda - length of waves (wave 1 and wave 2)
+## m - number of interference minimum
```
Comment (blackyblack) [inline symplyphysics/laws/optics/interference_minimum.py:13]: Mention somewhere that this number can be chosen arbitrarily to achieve the desired travel difference.

**Review by blackyblack — APPROVED** [review]
Sync with master to fix CI

File: symplyphysics/laws/optics/interference_minimum.py (hunk 10 above)
Comment (Voldoned) [reply symplyphysics/laws/optics/interference_minimum.py:13]: As far as I know $m=0; 1; 2; ...; \infty$

Comment (blackyblack) [reply symplyphysics/laws/optics/interference_minimum.py:13]: That's what I've send. You chose any positive integer number to pass into the law.

File: test/optics/interference_minimum_test.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    minimum_number = 2
+    wave_length = Quantity(550 * prefixes.nano * units.meters)
+    Args = namedtuple("Args", ["wave_length", "minimum_number"])
+    return Args(wave_length=wave_length, minimum_number=minimum_number)
+
+
+def test_basic_travel_difference(test_args):
+    result = minimum_law.calculate_travel_difference(test_args.wave_length, test_args.minimum_number)
+    assert SI.get_dimension_system().equivalent_dims(result.dimension, units.length)
+    result_travel_difference = convert_to(result, prefixes.nano * units.meters).evalf(5)
```
Comment (blackyblack) [inline test/optics/interference_minimum_test.py:24]: CI is failing somewhere around this line. Try to use brackets `(prefixes.nano * units.meters)`

Comment (Voldoned) [reply test/optics/interference_minimum_test.py:24]: It didn't work. And if we use only `units.meters`, this error occurred again.

File: symplyphysics/laws/optics/interference_minimum.py [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def print_law():
+    return print_expression(law)
+
+
+@validate_input(wave_length_=wave_length, number_minimum_=number_minimum)
+@validate_output(travel_difference)
+def calculate_travel_difference(wave_length_: Quantity, number_minimum_: int) -> Quantity:
+    solved = solve(law, wave_length, dict=True)[0][wave_length]
```
Comment (blackyblack) [inline symplyphysics/laws/optics/interference_minimum.py:32]: You solve for wave length, not for travel difference.

**Review by blackyblack — CHANGES_REQUESTED** [review]

File: symplyphysics/laws/optics/interference_minimum.py (hunk 12 above)
Comment (Voldoned) [reply symplyphysics/laws/optics/interference_minimum.py:32]: Thanks! This is the source of the error.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**